backend/uclapi/dashboard/webhook_views.py

```diff
@@ -0,0 +1,187 @@
+    payload = {
+        "service": "webhook",
+        "name": "challenge",
+        "challenge": ownership_challenge,
+        "verification_secret": verification_secret,
+    }
+
+    try:
+        req = requests.post(webhook_url, json=payload, timeout=3)
+        resp = req.json()
+    except (
+        ValueError,
+        requests.exceptions.Timeout,
+        requests.exceptions.ConnectionError
+    ):
+        return False
+    else:
+        if "challenge" not in resp.keys():
+            return False
+        return (
+            resp["challenge"] ==
+            ownership_challenge
+        )
+
+
+def edit_webhook(request):
+    if request.method != "POST":
+        response = JsonResponse({
+            "ok": False,
+            "message": (
+                "Request is not of method POST"
+            )
+        })
+        response.status_code = 400
+        return response
+
+    try:
+        app_id = request.POST["app_id"]
+        url = request.POST["url"]
+        siteid = request.POST["siteid"]
+        roomid = request.POST["roomid"]
+        contact = request.POST["contact"]
+        user_id = request.session["user_id"]
+    except KeyError:
+        response = JsonResponse({
+            "ok": False,
+            "message": (
+                "Request is missing parameters. Should have app_id"
+                ", url, siteid, roomid, contact"
+                " as well as a sessionid cookie"
+            )
+        })
+        response.status_code = 400
+        return response
+
+    if not user_owns_app(user_id, app_id):
+        response = JsonResponse({
+            "ok": False,
+            "message": (
+                "App does not exist or user is lacking permission."
+            )
+        })
+        response.status_code = 400
+        return response
+
+    app = App.objects.get(id=app_id)
+    webhook = app.webhook
+
+    if url != webhook.url:
+        if not is_url_safe(url):
+            response = JsonResponse({
+                "ok": False,
+                "message": (
+                    "Invalid URL"
+                )
+            })
+            response.status_code = 400
+            return response
+
+        if not verify_ownership(
+            url,
+            generate_secret(),
+            webhook.verification_secret
+        ):
+            response = JsonResponse({
+                "ok": False,
+                "message": (
+                    "Ownership of webhook can't be verified."
+                    "Make sure to follow the documentation: "
+                    "https://docs.uclapi.com/#challenge-event"
+                )
+            })
+            response.status_code = 400
+            return response
+
+        webhook.url = url
+        webhook.save()
+
+    webhook.siteid = siteid
+    webhook.roomid = roomid
+    webhook.contact = contact
+    webhook.enabled = True
+    webhook.save()
+
+    keen.add_event("Webhook edited", {
```

Comment:
This should use the delay task to run via Celery now that we have that configured